gammapy/modeling/fit.py

```diff
@@ -895,3 +899,82 @@ def _repr_html_(self):
             return self.to_html()
         except AttributeError:
             return f"<pre>{html.escape(str(self))}</pre>"
+
+
+class FitResults(collections.abc.Sequence):
+    def __init__(self, results, axis=None):
+        if axis and axis.nbin != len(results):
+            raise ValueError("Axis length does not correspond to number of results")
+
+        self.results = results
+        self.axis = axis
+
+    def __add__(self, other):
+        if isinstance(other, FitResult):
+            return FitResult([*self, other])
+        else:
+            raise TypeError(f"Invalid type: {other!r}")
+
+    def __getitem__(self, key):
+        if isinstance(key, np.ndarray) and key.dtype == bool:
+            return self.__class__(list(np.array(self.results)[key]))
+        else:
+            return self.__class__(self.results[key])
+
+    def __len__(self):
+        return len(self.results)
+
+    def success(self):
+        return [f.success for f in self.results]
+
+    def models(self):
+        return Models([model for f in self.results for model in f.models])
+
+    def covariance_results(self):
+        return [f.covariance_result for f in self.results]
+
+    def optimize_results(self):
+        return [f.optimize_result for f in self.results]
+
+    def select_by_interval(self, coord_min, coord_max):
+        if self.axis is None:
+            raise ValueError("No axis to convert coordinate to index!")
+
+        keymin = self.axis.coord_to_idx(coord_min)
+        keymax = self.axis.coord_to_idx(coord_max)
+
+        return self[keymin:keymax]
+
+    def write_models(self, path, **kwargs):
+        self.models().write(self.models, path, **kwargs)
+
+    def create_model_table(self):
+        col_names = []
+        col_unit = []
+
+        for par in self.models()[0].parameters.free_parameters:
+            col_names.append(par.name)
+            col_names.append(par.name + "_err")
+            unt = par.unit
+            if unt is u.Unit():
+                unt = ""
+            col_unit.append(unt)
+            col_unit.append(unt)
+
+        t = Table(names=col_names, units=col_unit)
+
+        for i in range(len(self)):
+            col_data = []
+
+            for name in self.models()[i].parameters.free_parameters.names:
+                col_data.append(self.models()[i].parameters[name].value)
+                col_data.append(self.models()[i].parameters[name].error)
+
+            t.add_row(col_data)
+
+        if isinstance(self.axis, TimeMapAxis):
+            t.add_columns(self.axis.to_gti().table.columns, indexes=[0, 0])
```

Comment:
Maybe we should have a `TimeMapAxis.to_table` directly

Reply:
Well yes, this way is quite convoluted...